features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Scopes/Delete.jsx

```diff
@@ -0,0 +1,159 @@
+            return op;
+        });
+        apiData.scopes = apiData.scopes.filter((scope) => {
+            return scope.name !== scopeName;
+        });
+        const setOpenLocal = setOpen; // Need to copy this to access inside the promise.then
+        const promisedUpdate = api.update(apiData);
+        promisedUpdate
+            .then(() => {
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Resources.Resources.api.scope.deleted.successfully',
+                    defaultMessage: 'API Scope Deleted successfully!',
+                }));
+                setOpenLocal(!open);
+            })
+            .catch((errorResponse) => {
+                console.error(errorResponse);
+                Alert.error(intl.formatMessage({
+                    id: 'Apis.Details.Resources.Resources.something.went.wrong.while.updating.the.api',
+                    defaultMessage: 'Error occurred while updating API',
+                }));
+            });
+    };
+
+    const runAction = (action) => {
+        if (action === 'yes') {
+            deleteScope();
+        } else {
+            setOpen(!open);
+        }
+    };
+    const { scopeName } = props;
+    return (
+        <div>
+            <Button onClick={toggleOpen}>
+                <Icon>delete_forever</Icon>
+                <FormattedMessage
+                    id='Apis.Details.Documents.Delete.document.delete'
+                    defaultMessage='Delete'
+                />
+            </Button>
+            <Dialog
+                open={open}
+                onClose={toggleOpen}
+                aria-labelledby='alert-dialog-title'
+                aria-describedby='alert-dialog-description'
+            >
+                <DialogTitle id='alert-dialog-title'>
+                    <FormattedMessage
+                        id='Apis.Details.Documents.Delete.document.listing.delete.confirm.title'
+                        defaultMessage='Are you sure to delete?'
+                    />
+                </DialogTitle>
+                <DialogContent>
+                    <DialogContentText id='alert-dialog-description'>
+                        <strong>{scopeName}</strong>
+                        <FormattedMessage
+                            id='Apis.Details.Documents.Delete.document.listing.delete.confirm.body'
+                            defaultMessage=' will be permernently deleted. Are you sure?'
+                        />
+                    </DialogContentText>
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={() => runAction('no')} color='primary'>
+                        <FormattedMessage
+                            id='Apis.Details.Documents.Delete.document.listing.delete.cancel'
+                            defaultMessage='Cancel'
+                        />
+                    </Button>
+                    <Button onClick={() => runAction('yes')} color='primary' autoFocus>
```

Comment:
I think it's better to use a button `name` here instead of creating a lambda function on the fly,

```suggestion
<Button name='cancel' onClick={runAction} color='primary' autoFocus>
```
and use the `event.target.name` to determine the `runAction` output.
That way is more clear I guess. and even it's easy to pick the button when writing tests 😄